confmap/confmap.go

```diff
@@ -189,6 +189,13 @@ func encoderConfig(rawVal interface{}) *encoder.EncoderConfig {
 	}
 }
 
+// case-sensitive version of the callback to be used in the MatchName property
+// of the DecoderConfig. The default for MatchEqual is to use strings.EqualFold,
+// which is case-insensitive.
+func caseSensitiveMatchName(a, b string) bool {
+	return a == b
```

Comment:
Should we trim the strings at least?

Reply:
Why?  To allow keys with spaces to work as well?
https://github.com/mitchellh/mapstructure/blob/v1.5.0/mapstructure.go#L275

Reply:
I agree with Raphael on this one, we should be strict.

Reply:
I think " foo" == "foo " in this case. Spaces in a config file will be hard to see if extra.